src/isar/services/readers/base_reader.py

```diff
@@ -47,16 +46,17 @@ def dict_to_dataclass(
             )
             return generated_dataclass
         except WrongTypeError as e:
-            logger.error(
-                f"A type of a input value does not match with a type of a data class field {e}"
-            )
-            return None
+            raise BaseReaderError(
+                "A type of a input value does not"
+                "match with a type of a data class field"
+            ) from e
         except MissingValueError as e:
-            logger.error(f"A value for a required field is not provided {e}")
-            return None
+            raise BaseReaderError("A value for a required field is not provided") from e
         except ValueError as e:
-            logger.error({e})
-            return None
+            raise BaseReaderError from e
         except Exception as e:
-            logger.error(f"Datastruct does not match expected struct {e}")
-            return None
+            raise BaseReaderError("Datastruct does not match expected struct") from e
```

Comment:
Not a strong preference, but I would consider removing the `try: except:` altogether and let a highet level catch the error. I think that the Error and message we get from the `dacite`-package is sufficient.